rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,361 @@
+
+- `username` (string): the username of the logged-in user
+- `badgeNumber` (string, optional): the badge number of the logged-in user
+- `stop` (string, optional): GTFS stop ID at which the logged-in user is located
+```json
+{
+  "username": "example",
+  "badgeNumber": "123456",
+  "stop": "place-lake"
+}
+
+or
+
+{
+  "username": "pswartz"
+}
+```
+It is represented as an object to provide future extensibility if needed.
+
+### Car
+Cars (individual parts of a train consist) will be described by their 4 digit number.
+```json
+{
+  "carNumber": "3800"
+}
+```
+
+### Operator
+Operators will be described by their badge number:
+```json
+{
+  "badgeNumber": "123"
+}
+```
+
+### Optional T
+A value that can either be an instance of T, or `null`. For example, Optional Car can be
+```json
+{ "carNumber": "3800"}
+# or 
+null
+```
+
+### Time
+A [GTFS Time](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#field-types).
+> Time in the HH:MM:SS format (H:MM:SS is also accepted). The time is measured from "noon minus 12h" of the service day (effectively midnight except for days on which daylight savings time changes occur). For times occurring after midnight, enter the time as a value greater than 24:00:00 in HH:MM:SS local time for the day on which the trip schedule begins.  
+> _Example: `14:30:00` for 2:30PM or `25:35:00` for 1:35AM on the next day._
+### Trip Key
+For scheduled trips, Glides does not have the same trip ID used by RTR. For added trips, while an ID is provided, it can be the case that an added trip from the Glides perspective is matches a GTFS trip, in which case RTR will us the GTFS trip ID in publishing the information.
+
+In order to reduce message duplication, a Trip Key is used to identify both added and scheduled trips.
+
+**Added Trip**
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `id` (string): unique value representing this trip for the given `serviceDate`
+```json
+`{"serviceDate": "2023-01-20", "id": "ADDED-123"}
+```
+**Scheduled Trip (departure time from origin)
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `departureTime` (Time): when the vehicle is scheduled to depart the given stop
+- `startStop` (string): GTFS stop ID the trip is departing from (either `location_type` 0 or 1)
+```json
+`{"serviceDate": "2023-01-20", "departureTime": "25:01:02", "startStop": "place-lake"}
+```
+**Scheduled Trip (arrival time at destination)
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `arrivalTime` (Time): when the vehicle is scheduled to arrive at the given stop
+- `endStop` (string): GTFS stop ID the trip is ending at (either `location_type` 0 or 1)
+```json
+`{"serviceDate": "2023-01-20", "arrivalTime": "09:10:11", "endStop": "place-lake"}
+```
+**Scheduled Trip (known GTFS trip ID)
+- `serviceDate` (RFC3999 date): the service date for the trip
+- `gtfsId` (string): GTFS trip ID
```

Comment:
I added it in response to https://github.com/mbta/technology-docs/pull/18/files/5f7b376f721f82b5fb6c0e65ff16bdd837ff5882#r1083578212, but as you note, it would require significant refactoring on the Glides side to actually use.

Reply:
Got it, that makes more sense, thanks!